pkg/kncloudevents/event_dispatcher_test.go

```diff
@@ -926,6 +927,128 @@ func TestSendEvent(t *testing.T) {
 		})
 	}
 }
+func TestDispatchMessageWithEventFormat(t *testing.T) {
+	t.Run("Binary format", testBinaryFormat)
+	t.Run("Json format", testJsonFormat)
+}
+
+func testBinaryFormat(t *testing.T) {
+	binary := kncloudevents.Binary
+	var wg sync.WaitGroup
+	ctx, _ := rectesting.SetupFakeContext(t)
+	ctx, cancel := context.WithCancel(ctx)
+	defer func() {
+		cancel()
+		time.Sleep(500 * time.Millisecond)
+	}()
+	oidcTokenProvider := auth.NewOIDCTokenProvider(ctx)
+	dispatcher := kncloudevents.NewDispatcher(eventingtls.NewDefaultClientConfig(), oidcTokenProvider)
+
+	eventToSend := test.FullEvent()
+	eventToSend.SetDataContentType("text/json")
+
+	// Destination setup
+	port := 8335
+	destinationEventsChan := make(chan cloudevents.Event, 10)
+	destinationReceivedEvents := make([]cloudevents.Event, 0, 10)
+	destinationHandler := eventingtlstesting.EventChannelHandler(destinationEventsChan)
+	destinationCA := eventingtlstesting.StartServer(ctx, t, port, destinationHandler, kncloudevents.WithDrainQuietPeriod(time.Millisecond))
+	destination := duckv1.Addressable{
+		URL:     apis.HTTPS(fmt.Sprintf("localhost:%d", port)),
+		CACerts: &destinationCA,
+	}
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for event := range destinationEventsChan {
+			destinationReceivedEvents = append(destinationReceivedEvents, event)
+		}
+	}()
+
+	// Set the event format
+	dispatcherOpts := kncloudevents.WithEventFormat(&binary)
+
+	// Send event
+	message := binding.ToMessage(&eventToSend)
+	info, err := dispatcher.SendMessage(ctx, message, destination, dispatcherOpts)
+	require.Nil(t, err)
+	require.Equal(t, 200, info.ResponseCode)
+
+	// Check received events
+	close(destinationEventsChan)
+	wg.Wait()
+
+	require.Len(t, destinationReceivedEvents, 1)
+	require.Equal(t, eventToSend.ID(), destinationReceivedEvents[0].ID())
+	require.Equal(t, eventToSend.Data(), destinationReceivedEvents[0].Data())
+
+	// Log the actual content type received
+	actualContentType := destinationReceivedEvents[0].DataContentType()
+	t.Logf("Received content type: %s", actualContentType)
+
+	// Additional checks to ensure the event format
+	require.Equal(t, "text/json", actualContentType)
+}
+
+func testJsonFormat(t *testing.T) {
```

Comment:
This function seems really similar to the `testBinaryFormat` function - is there some way we can make a more generic function that e.g. accepts the format we want and then runs the test for that format?

That way, we can maintain one function while getting two test cases covered